tokio-util/tests/task_join_queue.rs

```diff
@@ -0,0 +1,228 @@
+    let mut queue: JoinQueue<()> = JoinQueue::new();
+
+    for _ in 0..5 {
+        queue.spawn(futures::future::pending());
+    }
+    for _ in 0..5 {
+        queue.spawn(async {
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        });
+    }
+
+    // The join queue will now have 5 pending tasks and 5 ready tasks.
+    tokio::time::sleep(Duration::from_secs(2)).await;
+
+    queue.abort_all();
+    assert_eq!(queue.len(), 10);
+
+    let mut count = 0;
+    while let Some(res) = queue.join_next().await {
+        if count < 5 {
+            assert!(res.is_err());
+            assert!(res.unwrap_err().is_cancelled());
+        } else {
+            assert!(res.is_ok());
+        }
+        count += 1;
+    }
+    assert_eq!(count, 10);
+    assert_eq!(queue.len(), 0);
+}
+
+#[tokio::test]
+async fn test_join_queue_join_all() {
+    let mut queue = JoinQueue::new();
+    let mut senders = Vec::new();
+    for i in 0..5 {
+        let (tx, rx) = oneshot::channel::<()>();
+        senders.push(Some(tx));
+        queue.spawn(async move {
+            let _ = rx.await;
+            i
+        });
+    }
+    // Complete all tasks in reverse order
+    while let Some(tx) = senders.pop() {
+        let _ = tx.unwrap().send(());
+    }
+    let results = queue.join_all().await;
+    assert_eq!(results, vec![0, 1, 2, 3, 4]);
+}
+
+#[tokio::test]
+async fn test_join_queue_shutdown() {
+    let mut queue = JoinQueue::new();
+    let mut senders = Vec::new();
+
+    for _ in 0..5 {
+        let (tx, rx) = oneshot::channel::<()>();
+        senders.push(tx);
+        queue.spawn(async move {
+            let _ = rx.await;
+        });
+    }
+
+    queue.shutdown().await;
+    assert_eq!(queue.len(), 0);
+    while let Some(tx) = senders.pop() {
+        assert!(tx.is_closed());
+    }
+}
+
+#[tokio::test]
+async fn test_join_queue_with_manual_abort() {
+    let mut queue = JoinQueue::new();
+    let mut num_canceled = 0;
+    let mut num_completed = 0;
+    let mut senders = Vec::new();
+    for i in 0..16 {
+        let (tx, rx) = oneshot::channel::<()>();
+        senders.push(Some(tx));
+        let abort = queue.spawn(async move {
+            let _ = rx.await;
+            i
+        });
+
+        if i % 2 != 0 {
+            // abort odd-numbered tasks.
+            abort.abort();
+        }
+    }
+    // Complete all tasks in reverse order
+    while let Some(tx) = senders.pop() {
+        let _ = tx.unwrap().send(());
+    }
+    loop {
+        match queue.join_next().await {
+            Some(Ok(res)) => {
+                assert_eq!(res, num_completed * 2);
+                num_completed += 1;
+            }
+            Some(Err(e)) => {
+                assert!(e.is_cancelled());
+                num_canceled += 1;
+            }
+            None => break,
+        }
+    }
+
+    assert_eq!(num_canceled, 8);
+    assert_eq!(num_completed, 8);
+}
+
+#[tokio::test(flavor = "current_thread")]
```

Comment:
Emm, I think the default flavor is the `current_thread`. Is there any specific reason we set it explicitly?